mantidproject/mantid
# Disable unused WorkspaceIndex attribute for Q dependent functions

### Description of work
This PR ensures the `WorkspaceIndex` int property in the FunctionTreeView is disabled when it is not being used. When the property is not being used, it is defaulted to the maximum numeric limit for an int. This avoids confusing users who see an unexpectedly large WorkspaceIndex attribute in their function.

Fixes #29886

### To test:
1. Open the Inelastic data analysis GUI
2. Go to ConvFit 
3. Select a Q dependent function (TeixeiraWater)
4. click show full function 
5. The large WorkspaceIndex integer property for TeixeiraWater should be greyed out
6. The same should be the case for FickDiffusion and other Q dependent functions

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
